ethcore/src/client.rs

```diff
@@ -398,7 +401,7 @@ impl<V> Client<V> where V: Verifier {
 			}
 		}
 
-		if self.chain_info().best_block_hash != original_best {
+		if self.chain_info().best_block_hash != original_best && self.sealing_enabled.load(atomic::Ordering::Relaxed) {
```

Comment:
Avoid calling expensive `prepare_sealing` if we haven't already been asked for a work package from the miner.